webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/controllers/VolumeTracingController.scala

```diff
@@ -39,13 +43,45 @@ import com.scalableminds.webknossos.tracingstore.{
 import net.liftweb.common.{Box, Empty, Failure, Full}
 import play.api.i18n.Messages
 import play.api.libs.Files.TemporaryFile
-import play.api.libs.json.Json
+import play.api.libs.json.{Format, JsError, JsObject, JsResult, JsSuccess, JsValue, Json}
 import play.api.mvc.{Action, AnyContent, MultipartFormData, PlayBodyParsers}
 
 import java.io.File
 import java.nio.{ByteBuffer, ByteOrder}
 import scala.concurrent.ExecutionContext
 
+case class GetSegmentIndexParameters(
+    mag: Vec3Int,
+    cubeSize: Vec3Int, // Use the cubeSize parameter to map the found bucket indices to different size of cubes (e.g. reducing granularity with higher cubeSize)
+    additionalCoordinates: Option[Seq[AdditionalCoordinate]]
+)
+
+object GetSegmentIndexParameters {
+  implicit object GetSegmentIndexParametersFormat extends Format[GetSegmentIndexParameters] {
+    override def reads(json: JsValue): JsResult[GetSegmentIndexParameters] =
+      for {
+        magString <- (json \ "mag").validate[String]
```

Comment:
Could the frontend send number arrays here? Then we wouldn’t need a custom reads? Or is there a reason to use the stringified version? (same for CubeSize)